File: MK/10월 2주차/점프와순간이동.java
/**
 * 점프와 순간이동
 * https://programmers.co.kr/learn/courses/30/lessons/12980
 */
public class Solution {
    public int solution(int n) {
        int ans = 0;

        while (n != 0) {
            if (n%2==0) {
                n /= 2;
            } else {
                n--;
                ans++;
            }
        }

        return ans;
    }
}

File: BW/10월2주차/점프와순간이동.js
function solution(n) {
    var N=n;
    var K=0;
    
    while(N > 0){
        if(N%2 === 0) N /= 2; // 순간이동
        else{ // 한 칸 이동
            N--;
            K++;
        }
    }

    return K;
}